src/server/http/setup_connection.js

```diff
@@ -19,42 +25,42 @@ export default function (kbnServer, server, config) {
     }
   };
 
-  // enable tls if ssl key and cert are defined
-  if (config.get('server.ssl.key') && config.get('server.ssl.cert')) {
-    connectionOptions.tls = {
-      key: fs.readFileSync(config.get('server.ssl.key')),
-      cert: fs.readFileSync(config.get('server.ssl.cert')),
-      // The default ciphers in node 0.12.x include insecure ciphers, so until
-      // we enforce a more recent version of node, we craft our own list
-      // @see https://github.com/nodejs/node/blob/master/src/node_constants.h#L8-L28
-      ciphers: [
-        'ECDHE-RSA-AES128-GCM-SHA256',
-        'ECDHE-ECDSA-AES128-GCM-SHA256',
-        'ECDHE-RSA-AES256-GCM-SHA384',
-        'ECDHE-ECDSA-AES256-GCM-SHA384',
-        'DHE-RSA-AES128-GCM-SHA256',
-        'ECDHE-RSA-AES128-SHA256',
-        'DHE-RSA-AES128-SHA256',
-        'ECDHE-RSA-AES256-SHA384',
-        'DHE-RSA-AES256-SHA384',
-        'ECDHE-RSA-AES256-SHA256',
-        'DHE-RSA-AES256-SHA256',
-        'HIGH',
-        '!aNULL',
-        '!eNULL',
-        '!EXPORT',
-        '!DES',
-        '!RC4',
-        '!MD5',
-        '!PSK',
-        '!SRP',
-        '!CAMELLIA'
-      ].join(':'),
-      // We use the server's cipher order rather than the client's to prevent
-      // the BEAST attack
-      honorCipherOrder: true
-    };
+  // enable tlsOpts if ssl key and cert are defined
+  const useSsl = config.get('server.ssl.key') && config.get('server.ssl.cert');
+
+  // not using https? well that's easy!
+  if (!useSsl) {
+    server.connection(connectionOptions);
+    return;
   }
 
-  server.connection(connectionOptions);
+  // for https servers we have some extra settings
+  const tls = {
+    key: readFileSync(config.get('server.ssl.key')),
+    cert: readFileSync(config.get('server.ssl.cert')),
+
+    ciphers: tlsCiphers,
+    // We use the server's cipher order rather than the client's to prevent the BEAST attack
+    honorCipherOrder: true
+  };
+
+  server.connection({
+    ...connectionOptions,
+    tls: tls,
```

Comment:
Because we're passing the tls options to httpolyglot, I think you just need to pass `true` here.

http://hapijs.com/api#serverconnectionoptions